### DISCUSSION TITLE
Looking for a person to contact at NSA

### DISCUSSION THREAD

**Discussion by niconaus** [NationalSecurityAgency/ghidra#4117] [Q&A] [answered]
I'm a researcher at Virginia Tech, and I've developed a formal P-Code semantics. The work will be submitted to OOPSLA'22, and I was wondering if there is someone at NSA who works on Ghidra & P-Code, and would like to look at the paper, pre-publication.

The paper reports several issues with Ghidra and P-Code which would be nice to be aware of pre-publication, and we're also interested in getting their opinion.

**Comment by hedgeberg** [comment]
I suspect you're not going to get an answer, not because they don't care but because the pcode/sleigh/decompiler engine devs don't really do outside communication, as best as I can tell. I have a bug in the decompiler that has been present since release that I've been trying to reach out for help with fixing, and it hasn't worked. I've tried several routes of contact, including going through an intermediate friend, and it just hasn't worked out.

**Comment by niconaus** [comment, marked as answer]
I've been able to obtain contact information though VT's collaboration institute with the NSA.
